app/paypal.py

```
@@ -0,0 +1,242 @@
+async def get_paypal_public_key(cert_url: str) -> dsa.DSAPublicKey | rsa.RSAPublicKey | ec.EllipticCurvePublicKey | \
+                                                  ed25519.Ed25519PublicKey | ed448.Ed448PublicKey | \
+                                                  x25519.X25519PublicKey | x448.X448PublicKey | None:
+    if not cert_url.startswith("https://"):
+        return None
+    if not re.match("^https://(.+\.)?paypal.com/", cert_url):
+        return None
+    cert_cache_name = hashlib.sha256(cert_url.encode()).hexdigest()
+    try:
+        with open(f"/tmp/{cert_cache_name}.pem", "rb") as cert_file:
+            return x509.load_pem_x509_certificate(cert_file.read()).public_key()
+    except FileNotFoundError as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)